portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -46,19 +49,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     List<Post> posts = entityToList(results, request);
     
     response.setContentType("application/json;");
+    response.setCharacterEncoding("UTF-8");
     response.getWriter().println(convertToJson(posts));
   }
   
   // Posts information to the server to save as inputs from the data form
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Entity commentEntity = createEntity(request);
-
+  
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
 
     response.sendRedirect("/contact.html");
   }
+  
+  private String getLanguage(HttpServletRequest request) {
+    String languageChoice = request.getParameter("language");
```

Comment:
What happens if this fails?

Reply:
I created try/catch blocks for all of these instances using a NullPointerException (big umbrella). Let me know if you think that there's a better way to do exception handing here.